prow/crier/reporters/github/reporter.go

```diff
@@ -147,7 +148,12 @@ func (c *Client) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
 	}
 
 	// TODO(krzyzacy): ditch ReportTemplate, and we can drop reference to config.Getter
-	return []*v1.ProwJob{pj}, report.Report(c.gc, c.config().Plank.ReportTemplateForRepo(pj.Spec.Refs), *pj, c.config().GitHubReporter.JobTypesToReport)
+	err := report.Report(c.gc, c.config().Plank.ReportTemplateForRepo(pj.Spec.Refs), *pj, c.config().GitHubReporter.JobTypesToReport)
+	if err != nil && strings.Contains(err.Error(), "This SHA and context has reached the maximum number of statuses") {
```

Comment:
Just out of curious, where this error comes from? (search k/k gave me nothing), maybe we can compare with the actual error type?

Reply:
It comes from github and looks like this in the end:
```
"error setting status: status code 422 not one of [201], body: {\"message\":\"Validation Failed\",\"errors\":[{\"resource\":\"Status\",\"code\":\"custom\",\"message\":\"This SHA and context has reached the maximum number of statuses.\"}],\"documentation_url\":\"https://developer.github.com/v3/repos/statuses/#create-a-status\"}"
```